File: solutions/typescript/sieve/1/sieve.ts
export function primes(n: number): number[] {
  let sieve: boolean[] = Array(n+1).fill(true)
  let p = 2
  while (p <= n) {
    if (sieve[p]) {
        for (let i = 2; p*i <= n; i++) {
        sieve[p*i] = false;
        }
    }
    p += 1
  }
  let ans = []
  for (let i = 0; i <= n; i++) {
    if (sieve[i] && i >= 2) {
        ans.push(i)
    }
  }
  return ans
}

File: solutions/ocaml/sieve/1/sieve.ml
let rec sieve (arr : int list) : int list = 
  if List.is_empty arr
  then []
  else (List.hd arr) :: (sieve (List.filter (fun a -> a mod (List.hd arr) <> 0) (List.tl arr)))

let rec range (a : int) (b : int) : int list = 
  if a > b
  then []
  else a :: range (a + 1) b
let primes n = range 2 n |> sieve